src/ripple/app/misc/AMM.h

```diff
@@ -0,0 +1,160 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2022 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#ifndef RIPPLE_APP_MISC_AMM_H_INLCUDED
+#define RIPPLE_APP_MISC_AMM_H_INLCUDED
+
+#include <ripple/beast/utility/Journal.h>
+#include <ripple/protocol/Quality.h>
+#include <ripple/protocol/STAmount.h>
+#include <ripple/protocol/STArray.h>
+#include <ripple/protocol/STLedgerEntry.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/digest.h>
+
+namespace ripple {
+
+class ReadView;
+class ApplyView;
+class Sandbox;
+class STLedgerEntry;
+class NetClock;
+class STObject;
+class Rules;
+
+/** Calculate AMM account ID.
+ */
+template <typename... Args>
+AccountID
+calcAccountID(Args const&... args)
```

Comment:
I'm not sure `getAccountID(parentHash, groupHash)` is clearer than `accountID(parentHash, groupHash)`. I still prefer the no prefix version. I'm not worried about confusion between this function and a constructor. This function _does_ in fact act like a constructor, so that's not bad.

Reply:
Ok, will change.